### PR TITLE
Fixed array memory leak

### LINKED ISSUE #421: Memory leak in the array/string handling
This small contract produces a quick memory leak:

```solidity
contract C {
      string public symbol = "TRCT";
      string public name = "Tracto";
 
    function changeNameSymbol(string _name, string _symbol) {
          name = _name;
          symbol = _symbol; 
    }
}
```

I'm taking a look to this..

### PR DESCRIPTION
**Pull request by ggrieco-tob**
This fixes #421

### CODE REVIEW

File: lib/Echidna/Mutator.hs [hunk 1]
```diff
@@ -9,7 +9,7 @@ import qualified Data.ListLike as LL
 
 -- | A list of mutators to randomly select to perform a mutation of list-like values
 listMutators :: (LL.ListLike f i, MonadRandom m) => m (f -> m f)
-listMutators = fromList [(return, 1), (expandRandList, 10), (deleteRandList, 10), (swapRandList, 10)] 
+listMutators = fromList [(return, 1), (expandRandList, 10),  (deleteRandList, 10), (swapRandList, 10)]
```
Comment (incertia) [inline lib/Echidna/Mutator.hs:12]: rogue space

File: lib/Echidna/Mutator.hs [hunk 2]
```diff
@@ -63,11 +66,14 @@ swapAt xs i j =  let elemI = xs `LL.index` i
 swapRandList :: (LL.ListLike f i, MonadRandom m) => f -> m f
 swapRandList xs = if LL.null xs
                   then return xs
-                  else do k1 <- getRandomR (0, LL.length xs - 1 )
-                          k2 <- getRandomR (0, LL.length xs - 1 )
-                          return $ swapAt xs k1 k2
+                  else do i <- getRandomR (0, LL.length xs - 1 )
```
Comment (incertia) [inline lib/Echidna/Mutator.hs:69]: remove spaces before right parens

File: lib/Echidna/Mutator.hs [hunk 3]
```diff
@@ -26,19 +26,22 @@ replaceAt :: LL.ListLike f i => i -> f -> Int -> f
...
 expandAt :: LL.ListLike f i => f -> Int -> Int -> f
-expandAt xs i n = case LL.uncons xs of 
+expandAt xs k t = case LL.uncons xs of
                        Nothing     -> xs
-                       Just (y,ys) -> if i == 0
-                                      then LL.replicate n y `LL.append` ys
-                                      else LL.cons y (expandAt ys (i-1) n)
+                       Just (y,ys) -> if k == 0
+                                      then LL.replicate t y `LL.append` ys
+                                      else LL.cons y (expandAt ys (k-1) t)
```
Comment (incertia) [inline lib/Echidna/Mutator.hs:33]: add spaces in math

File: lib/Echidna/Mutator.hs [hunk 4]
```diff
@@ -26,19 +26,22 @@ replaceAt :: LL.ListLike f i => i -> f -> Int -> f
...
-expandRandList xs = if LL.null xs 
-                    then return xs
-                    else do k <- getRandomR (0, LL.length xs - 1)
-                            t <- getRandomR (1, max 32 (LL.length xs))
-                            return $ expandAt xs k t
+expandRandList xs
+              | l == 0    = return xs
+              | l >= 32   = return xs
+              | otherwise = do
+                             k <- getRandomR (0, l - 1)
```
Comment (incertia) [inline lib/Echidna/Mutator.hs:41]: indent 2 spaces

File: lib/Echidna/Mutator.hs [hunk 5]
```diff
@@ -26,19 +26,22 @@ replaceAt :: LL.ListLike f i => i -> f -> Int -> f
...
 
 
 expandRandList :: (LL.ListLike f i, MonadRandom m) => f -> m f
-expandRandList xs = if LL.null xs 
-                    then return xs
-                    else do k <- getRandomR (0, LL.length xs - 1)
-                            t <- getRandomR (1, max 32 (LL.length xs))
-                            return $ expandAt xs k t
+expandRandList xs
+              | l == 0    = return xs
```
Comment (incertia) [inline lib/Echidna/Mutator.hs:38]: guards should be indented 2 spaces

File: lib/Echidna/Mutator.hs [hunk 6]
```diff
@@ -23,22 +23,25 @@ mutateLL mn fs vs = do
...
-expandRandList xs = if LL.null xs 
-                    then return xs
-                    else do k <- getRandomR (0, LL.length xs - 1)
-                            t <- getRandomR (1, max 32 (LL.length xs))
-                            return $ expandAt xs k t
+expandRandList xs
+  | l == 0    = return xs
+  | l >= 32   = return xs
+  | otherwise = do
+                  k <- getRandomR (0, l - 1)
```
Comment (incertia) [inline lib/Echidna/Mutator.hs:41]:
```haskell
  | otherwise = do
    k <- getRandomR -- ...
  where -- ...
```

Comment (ggrieco-tob) [reply lib/Echidna/Mutator.hs]: Done.

**Pull request merged**